omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,236 @@
+	public String getIdentifierType(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType().getUuid();
+	}
+	
+	@PropertyGetter("source")
+	public String getSource(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getSource().getUuid();
+	}
+	
+	@PropertyGetter("uuid")
+	public String getUuid(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType().getUuid();
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType() + " - " + autoGenerationOption.getSource() + " - "
+		        + autoGenerationOption.getLocation();
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("uuid");
+		description.addProperty("identifierType");
+		description.addProperty("location");
+		description.addProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("location");
+		description.addProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public SimpleObject create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		String locationUuid = postBody.get("location").toString();
+		String manualEntryEnabled = postBody.get("manualEntryEnabled").toString();
+		String automaticGenerationEnabled = postBody.get("automaticGenerationEnabled").toString();
+		String sourceUuid = postBody.get("source").toString();
+		String identifierUuid = postBody.get("identifierType").toString();
+		AutoGenerationOption autoGenerationOption = new AutoGenerationOption();
+		
+		if ((locationUuid != null && manualEntryEnabled != null && automaticGenerationEnabled != null && sourceUuid != null
+		        && identifierUuid != null)) {
+			IdentifierSource identifierSource = identifierSourceService.getIdentifierSourceByUuid(sourceUuid);
+			Location location = Context.getLocationService().getLocationByUuid(locationUuid);
+			PatientIdentifierType patientIdentifierType = Context.getPatientService()
+			        .getPatientIdentifierTypeByUuid(identifierUuid);
+			autoGenerationOption.setLocation(location);
+			autoGenerationOption.setSource(identifierSource);
+			autoGenerationOption.setIdentifierType(patientIdentifierType);
+			autoGenerationOption.setManualEntryEnabled(Boolean.parseBoolean(manualEntryEnabled));
+			autoGenerationOption.setAutomaticGenerationEnabled(Boolean.parseBoolean(automaticGenerationEnabled));
+			
```

Comment:
What happens if these are not null but empty strings?

Reply:
adding more checks for that